server/container/guice/cassandra-guice/src/test/java/org/apache/james/CassandraJmapTestExtension.java

```diff
@@ -39,47 +38,117 @@
 import org.junit.jupiter.api.extension.ParameterContext;
 import org.junit.jupiter.api.extension.ParameterResolutionException;
 import org.junit.jupiter.api.extension.ParameterResolver;
+import org.junit.jupiter.api.function.ThrowingConsumer;
 import org.junit.rules.TemporaryFolder;
 
+import com.github.fge.lambdas.Throwing;
+import com.google.common.collect.ImmutableList;
 import com.google.inject.Module;
 
-class CassandraJmapTestExtension implements BeforeAllCallback, BeforeEachCallback, AfterAllCallback, AfterEachCallback, ParameterResolver {
+public class CassandraJmapTestExtension implements BeforeAllCallback, BeforeEachCallback, AfterAllCallback, AfterEachCallback, ParameterResolver {
+
+    public static class Builder {
+        public static Builder withDefaultModules() {
+            return withDefaultFromModules();
+        }
+
+        public static Builder withExtension(GuiceModuleTestExtension extension, Module... modules) {
+            return builder()
+                .modules(modules)
+                .extensions(ImmutableList.of(extension));
+        }
+
+        public static Builder withDefaultFromModules(Module... modules) {
+            return builder()
+                .modules(modules)
+                .extensions(EMBEDDED_ES_EXTENSION_ONLY);
+        }
+
+        private ImmutableList<Module> additionalModules;
+        private ImmutableList<GuiceModuleTestExtension> extensions;
+        private boolean ignoreEach;
+
+        public Builder() {
+            this.additionalModules = EMPTY_MODULES;
+            this.extensions = EMPTY_EXTENSIONS;
+            this.ignoreEach = false;
+        }
+
+        public Builder modules(Module... additionalModules) {
+            this.additionalModules = ImmutableList.copyOf(additionalModules);
+            return this;
+        }
+
+        public Builder extensions(ImmutableList<GuiceModuleTestExtension> extensions) {
+            this.extensions = extensions;
+            return this;
+        }
+
+        public Builder ignoreEach() {
```

Comment:
what does it mean ?

Reply:
please checkout #1759

Reply:
if you need to explain to me the intent of a piece of code, either I missed something obvious or the intent is not clear enough. In the second case, you then need to "express your intent better"(™) (cc @ddolcimascolo )

Reply:
sorry I will remove these lines in #1759, and added again when the code changes need it.